mkosi/config.py

```diff
@@ -4829,23 +4908,56 @@ def have_history(args: Args) -> bool:
     )
 
 
+def finalize_default_tools(main: dict[str, Any], *, resources: Path) -> Config:
+    context = ParseContext(resources)
+
+    for s in SETTINGS:
+        if s.scope == SettingScope.multiversal:
+            context.cli[s.dest] = copy.deepcopy(main[s.dest])
+
+    for k, v in main.items():
+        if dest := startswith(k, "tools_tree_"):
+            context.cli[dest] = copy.deepcopy(v)
+
+    context.cli["output_format"] = OutputFormat.directory
+
+    context.config = {
+        "image": "tools",
+        "directory": main["directory"],
+        "files": [],
+    }
+
+    context.config["environment"] = {
+        name: main["environment"][name]
+        for name in main.get("pass_environment", {})
```

Comment:
That should be a list, no?
```suggestion
        for name in main.get("pass_environment", [])
```

That being said, the whole comprehension could alternatively be written as
```suggestion
    context.config["environment"] = {
        name: main["environment"][name]
        for name in main.get("environment", {}).keys() & main.get("pass_environment", [])
    }
```
since dict keys support set operations.